python/ray/data/dataset_iterator.py

```diff
@@ -103,6 +104,31 @@ def iter_batches(
         """
         raise NotImplementedError
 
+    @abc.abstractmethod
+    def iter_rows(self, *, prefetch_blocks: int = 0) -> Iterator[Union[T, TableRow]]:
+        """Return a local row iterator over the dataset.
+
+        If the dataset is a tabular dataset (Arrow/Pandas blocks), dict-like mappings
+        :py:class:`~ray.data.row.TableRow` are yielded for each row by the iterator.
+        If the dataset is not tabular, the raw row is yielded.
```

Comment:
Nit: active voice, glossary term, and removal of unnecessary `:py:`
```suggestion
        If the dataset is a :term:`tabular dataset <Tabular Dataset>`, this iterator yields dict-like mappings
        :class:`~ray.data.row.TableRow` for each row. If the dataset isn't tabular, the raw row is yielded.
```

Reply:
The term "raw row" kinda confused me, but I can't think of a better alternative so it's probably fine.

Reply:
Defer discussion on verbatim code moves.